mii/deployment.py

```diff
@@ -66,86 +68,98 @@ def deploy(task,
         If deployment_type is `LOCAL`, returns just the name of the deployment that can be used to create a query handle using `mii.mii_query_handle(deployment_name)`
 
     """
+    if model_path is None and deployment_type == DeploymentType.LOCAL:
+        model_path = MII_MODEL_PATH_DEFAULT
+    elif model_path is None and deployment_type == DeploymentType.AML:
+        model_path = "model"
 
+    if not deployments and not all((model, task, deployment_name)):
+        assert deployment_tag is not None, "Deployment tag must be set when starting empty deployment"
+        create_score_file(deployment_tag=deployment_tag,
+                          deployment_type=deployment_type,
+                          deployments=None,
+                          model_path=model_path,
+                          port_map=None,
+                          lb_config=None)
+        return None
```

Comment:
This is a messy way to add this logic into the deployment function. We can refactor this as such:
```python
if deployment and any((model, task, deployment_name)):
    # Error out, it is ambiguous what we should do here
if all((model, task, deployment_name)):
    deployments.append(DeploymentConfig(...))
    if not deployment_tag:
        deployment_tag = deployment_name
assert deployment_tag, "deployment_tag must be set for multi-model"
for deployment in deployments:
    ...
...
create_score_file(...)
```

Reply:
I changed it and made a new function, let me know if you think it is better